agents/statsd/src/statsd_agent.cc

```diff
@@ -838,18 +841,17 @@ int StatsDAgent::config_handles() {
     return UV_EINVAL;
   }
 
+  if (connection_) {
+    connection_->close_and_delete();
+    connection_ = nullptr;
```

Comment:
I suppose this could be improved by checking if `connection_->endpoint()->type() == endpoint->type()`, and reusing it if that's the case.